packages/ui5-application-inquirer/src/prompts/prompts1.ts

```diff
@@ -232,7 +232,11 @@ export function getAddDeployConfigPrompt(
         // If the target directory is a CAP project then only offer `addDeployConfig (addToMta)` if an mta file is found
         when: async (answers: UI5ApplicationAnswers): Promise<boolean> => {
             mtaPath = (await getMtaPath(answers?.targetFolder || targetDir))?.mtaPath;
-            return !!(mtaPath && isCapProject) || !isCapProject;
+            const result = !!(mtaPath && isCapProject) || !isCapProject;
+            if (result && typeof addDeployConfigOptions?.hide === 'function') {
```

Comment:
Theres already a mechanism to implement hiding. Better not to mix it with the when condition?